curator-recipes/src/test/java/org/apache/curator/framework/recipes/shared/TestSharedCount.java

```diff
@@ -381,12 +385,118 @@ public void processResult(CuratorFramework client, CuratorEvent event) throws Ex
             }).forPath("/count");
             flushDone.await(5, TimeUnit.SECONDS);
 
-            Assert.assertEquals(2, numChangeEvents.get());
+            Assert.assertEquals(numChangeEvents.get(), 3);
```

Comment:
This means this PR includes semantic change of countHasChanged callback: after reconnecting, countHasChanged() is called.